### PR TITLE
Add functionality fetching all functions from a given schema

### LINKED ISSUE #58: Creating function class from PG system tables
## Background
To avoid clashes with existing functions they have to be dropped first. And to be dropped their signature has to be identified.

## Feature
The Postgres system tables need to be queried and the appropriate class instantiated.
Data to gather:
* schema (can be expected as an input)
* function name  (can be expected as an input, unless all function of the schema are to be dropped)
* IN parameters of the function (types suffice, but in order, as they are used with the name as the signature)

**NB!** Functions can be overloaded

The class then offers SQL entry to drop the function.

## Hint
```sql
SELECT
    n.nspname,
	A.rolname,
	T.typname AS returntype,
    p.*
    
FROM 
    pg_catalog.pg_namespace n JOIN 
    pg_catalog.pg_proc p ON p.pronamespace = n.oid JOIN
	pg_catalog.pg_authid A ON  p.proowner  = A.oid JOIN
	pg_catalog.pg_type T ON p.prorettype = T.oid
WHERE 	
	n.nspname NOT IN ('public', 'pg_catalog', 'pg_toast', 'information_schema') AND
	p.prokind = 'f'
ORDER BY n.nspname, p.proname
```

Depends on #56

### PR DESCRIPTION
**Pull request by jakipatryk**
Closes #58 ?

### CODE REVIEW

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala [hunk 1]
```diff
@@ -16,19 +16,61 @@
...
+      s"""
+         |SELECT
+         |  p.proname AS fn_name,
+         |  pg_catalog.pg_get_function_arguments(p.oid) AS in_and_out_arguments
+         |FROM
+         |  pg_catalog.pg_namespace n JOIN
+         |  pg_catalog.pg_proc p ON p.pronamespace = n.oid JOIN
+         |  pg_catalog.pg_type T ON p.prorettype = T.oid
+         |WHERE
+         |  n.nspname = '${schemaName.value}' AND
```
Comment (benedeki) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala:46]: Case sensitivity issue.

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala [hunk 2]
```diff
@@ -16,19 +16,61 @@
...
+                             schema: SchemaName,
+                             fnName: FunctionName,
+                             paramTypes: Seq[FunctionArgumentType]
+                           ) extends DBFunction {
+  override def sqlEntries: Seq[SQLEntry] = Seq(FunctionDrop(schema, fnName, paramTypes))
+}
+
+object DBFunctionFromPG {
+
+  def fetchAll(schemaName: SchemaName)
```
Comment (benedeki) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala:34]: Would name it `fetchAllOfSchema` or something like that.

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala [hunk 3]
```diff
@@ -16,19 +16,61 @@
...
+         |  p.proname AS fn_name,
+         |  pg_catalog.pg_get_function_arguments(p.oid) AS in_and_out_arguments
+         |FROM
+         |  pg_catalog.pg_namespace n JOIN
+         |  pg_catalog.pg_proc p ON p.pronamespace = n.oid JOIN
+         |  pg_catalog.pg_type T ON p.prorettype = T.oid
+         |WHERE
+         |  n.nspname = '${schemaName.value}' AND
+         |  p.prokind = 'f'
+         |ORDER BY n.nspname, p.proname;
```
Comment (benedeki) [inline ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala:48]: No ordering needed.

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala (hunk 1 above)
Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala:45]: WDYM exactly?

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala (hunk 2 above)
Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala]: done

File: ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala (hunk 3 above)
Comment (jakipatryk) [reply ultet/src/main/scala/za/co/absa/ultet/dbitems/DBFunctionFromPG.scala]: done

**Pull request merged**